Explorer/Assets/DCL/SDKComponents/AvatarAttach/Systems/AvatarAttachHandlerSystem.cs

```diff
@@ -0,0 +1,134 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.Throttling;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.Character.Components;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using DCL.SDKComponents.AvatarAttach.Components;
+using DCL.Utilities;
+using ECS.Abstract;
+using ECS.Groups;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Unity.Transforms.Components;
+using UnityEngine;
+
+namespace DCL.SDKComponents.AvatarAttach.Systems
+{
+    [UpdateInGroup(typeof(SyncedPostRenderingSystemGroup))]
+    [LogCategory(ReportCategory.AVATAR_ATTACH)]
+    [ThrottlingEnabled]
+    public partial class AvatarAttachHandlerSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private AvatarBase ownPlayerAvatarBase;
+
+        public AvatarAttachHandlerSystem(World world, WorldProxy globalWorld) : base(world)
+        {
+            globalWorld.Query(new QueryDescription().WithAll<PlayerComponent, AvatarBase>(), (ref PlayerComponent playerComponent, ref AvatarBase avatarBase) => { ownPlayerAvatarBase = avatarBase; });
+        }
+
+        protected override void Update(float t)
+        {
+            SetupAvatarAttachQuery(World);
+            UpdateAvatarAttachTransformQuery(World);
+            HandleComponentRemovalQuery(World);
+            HandleEntityDestructionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(AvatarAttachComponent))]
+        private void SetupAvatarAttach(in Entity entity, ref TransformComponent transformComponent, ref PBAvatarAttach pbAvatarAttach)
+        {
+            var component = new AvatarAttachComponent();
+
+            component.anchorPointTransform = GetAnchorPointTransform(pbAvatarAttach.AnchorPointId);
+
+            ApplyAnchorPointTransformValues(transformComponent, component);
+            transformComponent.UpdateCache();
+            World.Set(entity, transformComponent);
+
+            World.Add(entity, component);
+        }
+
+        [Query]
+        private void UpdateAvatarAttachTransform(in Entity entity, ref PBAvatarAttach pbAvatarAttach, ref AvatarAttachComponent avatarAttachComponent, ref TransformComponent transformComponent)
+        {
+            if (pbAvatarAttach.IsDirty)
+            {
+                avatarAttachComponent.anchorPointTransform = GetAnchorPointTransform(pbAvatarAttach.AnchorPointId);
+                World.Set(entity, avatarAttachComponent);
+            }
+
+            if (ApplyAnchorPointTransformValues(transformComponent, avatarAttachComponent))
+            {
+                transformComponent.UpdateCache();
+                World.Set(entity, transformComponent);
+            }
+        }
+
+        [Query]
+        [All(typeof(AvatarAttachComponent))]
+        [None(typeof(PBAvatarAttach), typeof(DeleteEntityIntention))]
+        private void HandleComponentRemoval(in Entity entity)
```

Comment:
Replace it in favor of the bulk deletion: introduce the query description manually and call
```
/// <summary>
    ///     An efficient method to remove one component from <see cref="T:Arch.Core.Entity" />s matching a <see cref="T:Arch.Core.QueryDescription" />.
    ///     No <see cref="T:Arch.Core.Entity" />s are recopied which is much faster.
    /// </summary>
    /// <remarks>Causes a structural change.</remarks>
    /// <param name="queryDescription">The <see cref="T:Arch.Core.QueryDescription" /> which specifies which <see cref="T:Arch.Core.Entity" />s will be targeted.</param>
    [SkipLocalsInit]
    [StructuralChange]
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public unsafe void Remove<T>(in QueryDescription queryDescription)
```

Reply:
Done!